### PR TITLE
Make EXE executable on Posix.

### PR DESCRIPTION
**Pull request by gejohnston**
Set the executable bit when 'zowe daemon enable' creates the EXE file.

### CODE REVIEW

**Comment by gejohnston** [comment]
Confirmed the behavior on Linux with a build from source.

File: packages/cli/CHANGELOG.md
Comment (awharn) [inline packages/cli/CHANGELOG.md:7]: Linux and Mac

**Review by awharn — COMMENTED** [review]
One small comment. 😄

File: packages/cli/src/daemon/enable/Enable.handler.ts
Comment (t1m0thyj) [inline packages/cli/src/daemon/enable/Enable.handler.ts:197]: Curious about this: 
`sysInfo.platform == "linux" || sysInfo.platform == "darwin"`
vs this:
`sysInfo.platform !== "win32"`

It probably doesn't matter, as I don't we think support of the other systems that may be returned by [process.platform](https://nodejs.org/api/process.html#processplatform): `aix`, `freebsd`, `openbsd`, `sunos`

File: packages/cli/CHANGELOG.md
Comment (gejohnston) [reply packages/cli/CHANGELOG.md:7]: Duh !!!
Thanks.

File: packages/cli/src/daemon/enable/Enable.handler.ts
Comment (awharn) [reply packages/cli/src/daemon/enable/Enable.handler.ts:197]: That might be shorter. I just suggested that line because Windows is working right now. It might be that we don't even need to do a check and can just set the permission bits to 0o750 for every OS.

**Review by zFernand0 — APPROVED** [review]

**Review by awharn — APPROVED** [review]

**Review by t1m0thyj — COMMENTED** [review]
Before I approve, not sure why some integration tests failed in the Jenkins pipeline?

**Comment by gejohnston** [comment]
I don't know why the integration tests are failing either.  I just started looking into it.

**Comment by awharn** [comment]
Well, we just set the executable bit, so maybe it thinks the zowe executable is a bash script now?

**Comment by gejohnston** [comment]
I just now manually ran the integration tests successfully on Windows and in a Linux VM, so I still do not know why the Jenkins pipeline is unhappy.

**Comment by awharn** [comment]
@gejohnston I just took a look at the logic for that test and noticed something. The Linux zowe "executable" (the NPM bin) is a symlink to main.js (which statSync.size says is 2963 because it follows the symlink). I don't know if that might be screwing up the integration test, but it might be a lead.

**Comment by gejohnston** [comment]
I am not sure why these tests just started failing now, but I will increase the max size test for a script from 2000 to 5000 and see what happens.

**Comment by awharn** [comment]
The only other thing I can think if is maybe we need to await rimraf or call rimraf.sync in the beforeEach?

**Comment by gejohnston** [comment]
Once we set the -x bit, it changed the nature of the error that occurs when daemon-enable runs a fake EXE. The test-case failed due the change in error output. By removing a test that expects stderr to be empty, the tests pass. I do not know why the tests ran on Linux manually, but failed in the Jenkins pipeline. The test was unnecessary, so removing it seemed reasonable.

Now, both Github Actions and Jenkins are successful. This PR can be merged after it has been approved.

**Review by t1m0thyj — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

**Review by awharn — APPROVED** [review]

**Pull request merged**
